scripts/pre_commit_linter.py

```diff
@@ -666,15 +668,53 @@ def _check_import_order(all_files):
 
     return summary_messages
 
+
+def _check_def_spacing(all_files):
+    """This function is used to check the number of blank lines
+    left above each class, function and method defintion.
+    """
+    print 'Starting def-spacing checks'
+    print '----------------------------------------'
+    print ''
+    pycodestyle_config_path = os.path.join(os.getcwd(), 'tox.ini')
+    summary_messages = []
+    # Selecting only Python files to check for errors
+    all_files = [
```

Comment:
`files_to_check` (since you already have `all_files` in the method signature)

Reply:
This list comprehension is also quite a handful, you might want to split it up. Keep in mind that the comprehension regenerates the latter for loop more than it needs to.

Reply:
To keep up with the consistency with the rest of the code, I went with list comprehension. Should  I split it up irrestpective of this?  
Thanks!

Reply:
You can use list comprehension, but you have a list comprehension within a list comprehension, the latter one is needlessly reworked every cycle :)

Reply:
Okay, got it! Thanks :)

Reply:
Hello @kevinlee12!  
```python
for filename in all_files:
        if (
                not any(fnmatch.fnmatch(filename, pattern)
                        for pattern in EXCLUDED_PATHS)
                and filename.endswith('.py')):
            files_to_check.append(filename)  
```
Is this good to go?  
Thanks!

Reply:
oops, my apologizes, I completely misinterpreted this bit of code. your previous implement is correct, but do rename the assignment.

Reply:
No problem! I'll rename the assignment then! Thanks!